src/main/java/org/broadinstitute/consent/http/service/dao/UserServiceDAO.java

```diff
@@ -42,4 +45,34 @@ public void updateInstitutionAndClearLibraryCardForUser(Integer userId, Integer
           libraryCardDAOT.deleteAllLibraryCardsByUser(userId);
         });
   }
+
+  public User createUser(User user) {
+    return jdbi.inTransaction(
+        transactionHandle -> {
+          if (user.getRoles() == null || user.getRoles().isEmpty()) {
+            throw new IllegalArgumentException("At least one role must be specified for the user.");
+          }
+          UserDAO userDAOT = transactionHandle.attach(UserDAO.class);
+          UserRoleDAO userRoleDAOT = transactionHandle.attach(UserRoleDAO.class);
+          LibraryCardDAO libraryCardDAOT = transactionHandle.attach(LibraryCardDAO.class);
+          Integer userId =
+              userDAOT.insertUser(
+                  user.getEmail(), user.getDisplayName(), user.getInstitutionId(), new Date());
+          userDAOT.updateEmailPreference(userId, user.getEmailPreference());
+          for (UserRole role : user.getRoles()) {
+            userRoleDAOT.insertSingleUserRole(role.getRoleId(), userId);
+          }
+          LibraryCard libraryCard = libraryCardDAOT.findLibraryCardByUserEmail(user.getEmail());
```

Comment:
If we eventually refactor how we allow signing officials to issue Library Cards (so they also create the underlying user) it simplifies this code as well.